php_phongo.c

```diff
@@ -953,6 +952,13 @@ static bool php_phongo_apply_options_to_uri(mongoc_uri_t *uri, bson_t *options T
 			continue;
 		}
 
+        if (!strcasecmp(key, MONGOC_URI_READPREFERENCE)) {
```

Comment:
That's why I'd asked about the right context, because I felt that we should use this validation within the `php_phongo_apply_rp_options_to_uri()` context, but I get confused with
> but I think the general php_phongo_apply_options_to_uri() function may be best since ...

so I decided to go with `php_phongo_apply_options_to_uri`, However I will fix this, but in this case we will pass the key literally in the exception message [readPreference], since we can not use `key` variable, and I think that it will not be proper to define it `const char *key = bson_iter_key(&iter);` for only that use